runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+	"OperationGreater":           OperationGreater,
+	"OperationLessEqual":         OperationLessEqual,
+	"OperationGreaterEqual":      OperationGreaterEqual,
+	"OperationPlus":              OperationPlus,
+	"OperationMinus":             OperationMinus,
+	"OperationMul":               OperationMul,
+	"OperationDiv":               OperationDiv,
+	"OperationMod":               OperationMod,
+	"OperationNegate":            OperationNegate,
+	"OperationNilCoalesce":       OperationNilCoalesce,
+	"OperationMove":              OperationMove,
+	"OperationCast":              OperationCast,
+	"OperationFailableCast":      OperationFailableCast,
+	"OperationForceCast":         OperationForceCast,
+	"OperationBitwiseOr":         OperationBitwiseOr,
+	"OperationBitwiseXor":        OperationBitwiseXor,
+	"OperationBitwiseAnd":        OperationBitwiseAnd,
+	"OperationBitwiseLeftShift":  OperationBitwiseLeftShift,
+	"OperationBitwiseRightShift": OperationBitwiseRightShift,
+}
+
+var statementKindMap = map[string]func() Statement{
+	"AssignmentStatement":        func() Statement { return &AssignmentStatement{} },
+	"BreakStatement":             func() Statement { return &BreakStatement{} },
+	"CompositeDeclaration":       func() Statement { return &CompositeDeclaration{} },
+	"ContinueStatement":          func() Statement { return &ContinueStatement{} },
+	"EmitStatement":              func() Statement { return &EmitStatement{} },
+	"ExpressionStatement":        func() Statement { return &ExpressionStatement{} },
+	"ForStatement":               func() Statement { return &ForStatement{} },
+	"FunctionDeclaration":        func() Statement { return &FunctionDeclaration{} },
+	"IfStatement":                func() Statement { return &IfStatement{} },
+	"ImportDeclaration":          func() Statement { return &ImportDeclaration{} },
+	"InterfaceDeclaration":       func() Statement { return &InterfaceDeclaration{} },
+	"PragmaDeclaration":          func() Statement { return &PragmaDeclaration{} },
+	"ReturnStatement":            func() Statement { return &ReturnStatement{} },
+	"SpecialFunctionDeclaration": func() Statement { return &SpecialFunctionDeclaration{} },
+	"SwapStatement":              func() Statement { return &SwapStatement{} },
+	"SwitchStatement":            func() Statement { return &SwitchStatement{} },
+	"TransactionDeclaration":     func() Statement { return &TransactionDeclaration{} },
+	"VariableDeclaration":        func() Statement { return &VariableDeclaration{} },
+	"WhileStatement":             func() Statement { return &WhileStatement{} },
+}
+
+var transferOperationMap = map[string]TransferOperation{
+	"TransferOperationUnknown":    TransferOperationUnknown,
+	"TransferOperationCopy":       TransferOperationCopy,
+	"TransferOperationMove":       TransferOperationMove,
+	"TransferOperationMoveForced": TransferOperationMoveForced,
+}
+
+var typeNameMap = map[string]func() Type{
+	"ConstantSizedType": func() Type { return &ConstantSizedType{} },
```

Comment:
What is the reason for the value being a function here? Can the value type of the map just be Expression?